ssl/t1_lib.c

```diff
@@ -181,22 +184,56 @@ static const uint16_t eccurves_default[] = {
     30,                      /* X448 (30) */
     25,                      /* secp521r1 (25) */
     24,                      /* secp384r1 (24) */
+#ifndef OPENSSL_NO_CNSM
+    249,                     /* sm2 (default 249, use set_sm2_group_id_custom(int value) to config) */
+#endif
+};
+
+#ifndef OPENSSL_NO_CNSM
+/* The default sm2 curves */
+static  uint16_t eccurves_default_sm2[11] = {
+    249,                      /* tassl default (249) */
 };
 
+uint16_t SM2_group_id_custom = 249;
+static int sm2_curve_config_index = 0;
+int set_sm2_group_id_custom(uint16_t value)
+{
+    if(value > 0){
+        SM2_group_id_custom = value;
+        memcpy(eccurves_default_sm2+1+sm2_curve_config_index++, &SM2_group_id_custom, sizeof(uint16_t));
+        return 1;
+    }
+
+    return 0;	
+}
+
+#endif
+
 static const uint16_t suiteb_curves[] = {
     TLSEXT_curve_P_256,
     TLSEXT_curve_P_384
 };
 
 const TLS_GROUP_INFO *tls1_group_id_lookup(uint16_t group_id)
 {
+#ifndef OPENSSL_NO_CNSM
+    if(SM2_group_id_custom != 249 && SM2_group_id_custom > 0){
+    	return &nid_list[248];
+    }
+#endif
+
     /* ECC curves from RFC 4492 and RFC 7027 */
     if (group_id < 1 || group_id > OSSL_NELEM(nid_list))
         return NULL;
     return &nid_list[group_id - 1];
 }
 
+#ifndef OPENSSL_CNSM
+uint16_t tls1_nid2group_id(int nid)
+#else
 static uint16_t tls1_nid2group_id(int nid)
+#endif
```

Comment:
I have seen this construct in several places, and find it unnecessary. If you need to remove `static` from a function for some reason, that's OK. But please do double check that it's actually necessary.